other/upgrade_2-1_postgresql.sql

```diff
@@ -1814,33 +1910,37 @@ $request = upgrade_query("
 
 	while ($row = $smcFunc['db_fetch_assoc']($request))
 	{
-		$inserts[] = "($row[id_group], 'profile_blurb_own', $row[add_deny])";
-		$inserts[] = "($row[id_group], 'profile_displayed_name_own', $row[add_deny])";
-		$inserts[] = "($row[id_group], 'profile_forum_own', $row[add_deny])";
-		$inserts[] = "($row[id_group], 'profile_website_own', $row[add_deny])";
-		$inserts[] = "($row[id_group], 'profile_signature_own', $row[add_deny])";
+		$inserts[] = array($row['id_group'], 'profile_blurb_own', $row['add_deny']);
+		$inserts[] = array($row['id_group'], 'profile_displayed_name_own', $row['add_deny']);
+		$inserts[] = array($row['id_group'], 'profile_forum_own', $row['add_deny']);
+		$inserts[] = array($row['id_group'], 'profile_website_own', $row['add_deny']);
+		$inserts[] = array($row['id_group'], 'profile_signature_own', $row['add_deny']);
 	}
 
 	$smcFunc['db_free_result']($request);
 
 	if (!empty($inserts))
 	{
-		foreach ($inserts as $insert)
-		{
-			upgrade_query("
-				INSERT INTO {$db_prefix}permissions
-					(id_group, permission, add_deny)
-				VALUES
-					" . $insert
-			);
-		}
+		$smcFunc['db_insert']('replace',
+			'{db_prefix}permissions',
+			array('id_group' => 'int', 'permission' => 'string', 'add_deny' => 'int'),
+			$inserts,
+			array('id_group', 'permission')
+		);
 	}
 ---}
 ---#
 
 /******************************************************************************/
 --- Upgrading PM labels...
 /******************************************************************************/
+---# upgrade check
+---{
+$table_columns = $smcFunc['db_list_columns']('{db_prefix}pm_recipients');
```

Comment:
I don't think we should do this test. The "Moving label info to new tables and updating rules" sub-step contains lots of different things that it needs to check and manipulate—much more than this simple test can account for.